torchdata/datapipes/iter/util/dataframemaker.py

```diff
@@ -39,3 +41,40 @@ def __new__(
         batch_dp = source_dp.batch(dataframe_size)
         df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))
         return df_dp
+
+
+@functional_datapipe("load_parquet_as_df")
+class ParquetDFIterDataPipe(IterDataPipe):  # IterDataPipe[torcharrow.IDataFrame[T_co]]
+    r"""
+    Iterable DataPipe that takes in paths to Parquet files and return a TorchArrow DataFrame for each row group
+    within a Parquet file.
+
+    Args:
+        source_dp: source DataPipe containing paths to the Parquet files
+        columns: List of str that specifies the column names of the DataFrame
+        use_threads: if True, Parquet reader will perform multi-threaded column reads
+        dtype: specify the TorchArrow dtype for the DataFrame
+        device: specify the device on which the DataFrame will be stored
+    """
+
+    def __init__(
+        self,
+        source_dp: IterDataPipe[str],
+        columns: Optional[List[str]] = None,
+        use_threads: bool = False,
+        dtype=None,  # Optional[torcharrow.dtypes.DType]
+        device: str = "",
+    ):
+        self.source_dp = source_dp
+        self.columns = columns
+        self.use_threads = use_threads
+        self.dtype = dtype
+        self.device = device
+
+    def __iter__(self):
+        for path in self.source_dp:
+            parquet_file = parquet.ParquetFile(path)
+            num_row_groups = parquet_file.num_row_groups
+            for i in range(num_row_groups):
+                row_group = parquet_file.read_row_group(i, columns=self.columns, use_threads=self.use_threads)
```

Comment:
We can potentially allow multiple row groups within each DataFrame. A typical row group seems to be 0.5-1GB (according to a quick Google search)